### PR TITLE
gitignore: adding new generate files

### PR DESCRIPTION
**Pull request by yfguo**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -389,6 +389,7 @@ Makefile.am-stamp
 /maint/createhtmlindex
 /maint/extractcvars
 /maint/cvardirs
+/maint/f77tof90
```
Comment (raffenet) [inline .gitignore]: Are you seeing this file generated? It should be moved to `test/mpi/maint`.

Comment (wesbland) [reply .gitignore]: @yfguo - I think this just comes from working on a branch from before the patch that moved this and then changing to a branch from after. I see this a lot.

File: .gitignore [hunk 2]
```diff
@@ -604,3 +605,7 @@ Makefile.am-stamp
 
 # /src/mpid/ch4/include
 /src/mpid/ch4/include/netmodpre.h
+
+# /src/mpid/ch4/shm/posix
+src/mpid/ch4/shm/posix/eager/include/posix_eager_pre.h
+src/mpid/ch4/shm/posix/posix_eager_array.c
```
Comment (wesbland) [inline .gitignore]: Sorry I missed these. I almost never build in-tree. Should these be in `src/mpid/ch4/.gitignore`?

Comment (raffenet) [reply .gitignore]: Most likely they should. In that case, we should move the remaining ch4 related files in the top-level `.gitignore` as well.

**Pull request merged**